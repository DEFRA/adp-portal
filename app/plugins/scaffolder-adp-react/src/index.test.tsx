import React from 'react';
import { getComponentData } from '@backstage/core-plugin-api';
import { DeliveryProjectPickerFieldExtension } from './index';
import {
  DeliveryProjectPicker,
  DeliveryProjectPickerSchema,
} from './components/DeliveryProjectPicker/DeliveryProjectPicker';

describe('DeliveryProjectPickerFieldExtension', () => {
  it('Should be match the expected component data', () => {
    const data = getComponentData(
      <DeliveryProjectPickerFieldExtension />,
      'scaffolder.extensions.field.v1',
    );
    expect(data).toEqual({
      component: DeliveryProjectPicker,
      name: 'DeliveryProjectPicker',
      schema: DeliveryProjectPickerSchema,
    });
  });
});
