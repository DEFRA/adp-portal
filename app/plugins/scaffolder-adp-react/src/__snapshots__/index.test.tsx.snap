// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryProjectPickerFieldExtension Should be match the expected component data 1`] = `
{
  "component": [Function],
  "name": "DeliveryProjectPicker",
  "schema": {
    "returnValue": {
      "$schema": "http://json-schema.org/draft-07/schema#",
      "type": "string",
    },
    "uiOptions": {
      "$schema": "http://json-schema.org/draft-07/schema#",
      "additionalProperties": false,
      "properties": {},
      "type": "object",
    },
  },
}
`;
